src/go/pt-k8s-debug-collector/dumper/dumper.go

```diff
@@ -0,0 +1,355 @@
+package dumper
+
+import (
+	"archive/tar"
+	"bytes"
+	"compress/gzip"
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+	"log"
+	"os"
+	"os/exec"
+	"strings"
+	"time"
+
+	"github.com/pkg/errors"
+	corev1 "k8s.io/api/core/v1"
+)
+
+// Dumper struct is for dumping cluster
+type Dumper struct {
+	cmd       string
+	resources []string
+	namespace string
+	location  string
+	errors    string
+	mode      int64
+	crType    string
+}
+
+// New return new Dumper object
+func New(location, namespace, resource string) Dumper {
+	resources := []string{
+		"pods",
+		"replicasets",
+		"deployments",
+		"statefulsets",
+		"replicationcontrollers",
+		"events",
+		"configmaps",
+		"secrets",
+		"cronjobs",
+		"jobs",
+		"podsecuritypolicies",
+		"poddisruptionbudgets",
+		"perconaxtradbbackups",
+		"perconaxtradbclusterbackups",
+		"perconaxtradbclusterrestores",
+		"perconaxtradbclusters",
+		"clusterrolebindings",
+		"clusterroles",
+		"rolebindings",
+		"roles",
+		"storageclasses",
+		"persistentvolumeclaims",
+		"persistentvolumes",
+	}
+	if len(resource) > 0 {
```

Comment:
Do we need to check for duplicates?

Reply:
no need in this check, in final tar file there would not be any duplicates